src/utils/decorators.spec.ts

```diff
@@ -0,0 +1,509 @@
+/**
+ * Copyright 2023 Google LLC.
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as chai from 'chai';
+
+import {feed, eat, pantry} from './decorators.js';
+
+const expect = chai.expect;
+
+const SymbolSym = Symbol('symbol');
+const SymbolSym2 = Symbol('symbol2');
+
+declare module './decorators.js' {
+  interface InjectableRegistry {
+    [SymbolSym]: symbol;
+    [SymbolSym2]: symbol;
+  }
+}
+
+describe('Decorators', () => {
+  describe('inject: field, provides: field, consumes: field', () => {
+    it('should work', () => {
+      class A {
+        @pantry(SymbolSym)
```

Comment:
I think we should have a bigger discussion about this and wait for Thiago to decide on this. I personally not much in favour of using decorators for injecting members as I find it harder to read and more verbose than the regular parameter passing. I think we should discuss it more in the daily (is there a dog about pros/cons) and also consider if this will work in Node 16+.

Reply:
Sure, we can wait on Thiago.

My two cents: Sure, the approach can be a little verbose, but I can't agree that it's harder to read. A simple proof of this is by comparing the following two files:

Without injection:
https://github.com/GoogleChromeLabs/chromium-bidi/blob/41d16155105b706383ca6d9d6651443bd697b91c/src/bidiMapper/CommandProcessor.ts#L60

With injection:
https://github.com/GoogleChromeLabs/chromium-bidi/blob/9c382d37e89a4137459a72073f009f00edcbd5a9/src/bidiMapper/CommandProcessor.ts#L53

It's very clear from injection what is going into those objects without even looking at the definition of the constructor. If you consider line count, without injection is also about 20% more verbose and that's with only partial injection.